pkg/vcs/client.go

```diff
@@ -38,13 +40,28 @@ func NewClient(conn *model.Connector) (*scm.Client, error) {
 	return client, nil
 }
 
-func NewClientFromURL(driver, url, token string) (*scm.Client, error) {
+func NewClientFromURL(driver, rawURL, token string) (*scm.Client, error) {
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+	// TODO support reverse proxy for self-hosted server.
+	server := u.Scheme + "://" + u.Host
+
 	switch driver {
 	case github.Driver:
-		return github.NewClientFromURL(url, token)
+		return github.NewClientFromURL(server, token)
 	case gitlab.Driver:
-		return gitlab.NewClientFromURL(url, token)
+		return gitlab.NewClientFromURL(server, token)
 	}
 
 	return nil, fmt.Errorf("unsupported SCM driver %q", driver)
 }
```

Comment:
can we return a scm.Client with User-Agent directly?

Reply:
updated